lib/stellar_core_commander/docker_process.rb

```diff
@@ -206,18 +206,44 @@ def shared_history_volume
     end
 
     Contract None => String
-    def history_commands
-      if use_s3
-        <<-EOS.strip_heredoc
-          HISTORY_GET=aws s3 --region #{@s3_history_region} cp #{@s3_history_prefix}/%s/{0} {1}
-          HISTORY_PUT=aws s3 --region #{@s3_history_region} cp {0} #{@s3_history_prefix}/%s/{1}
-        EOS
+    def history_get_command
+      cmds = {}
+      @quorum.each do |q|
+        if q == @name
+          next
+        end
+        cmd = ""
+        if SPECIAL_PEERS.has_key? q
+          cmd = SPECIAL_PEERS[q][:get]
+        elsif use_s3
+          cmd = "aws s3 --region #{@s3_history_region} cp #{@s3_history_prefix}/%s/{0} {1}"
+        else
+          cmd = "cp /history/%s/{0} {1}"
+        end
+        cmds[cmd] = ()
+      end
+      if cmds.size != 1
+        raise "Conflicting get commands: #{cmds.keys}"
+      end
+      <<-EOS.strip_heredoc
+        HISTORY_GET=#{cmds.keys.first}
+      EOS
+    end
+
+    Contract None => String
+    def history_put_commands
+      if @quorum.any? {|q| SPECIAL_PEERS.has_key? q}
+        ""
```

Comment:
Why omit history altogether in the special peer case? Seems like it couldn't hurt to have the local history.